File: src/order-service/src/Orders.Core/PublicEvents/OrderCreatedEventV1.cs
// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2024 Datadog, Inc.

using System.Text.Json.Serialization;

namespace Orders.Core.PublicEvents;

public record OrderCreatedEventV1
{
    [JsonPropertyName("orderNumber")]
    public string OrderNumber { get; set; } = "";

    [JsonPropertyName("userId")]
    public string UserId { get; set; } = "";

    [JsonPropertyName("products")]
    public string[] Products { get; set; } = [];
}

File: src/inventory-service/inventory-api/src/test/java/com/inventory/api/driver/OrderCreatedEventV1.java
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2024 Datadog, Inc.
 */

package com.inventory.api.driver;

import java.util.List;

public class OrderCreatedEventV1 {
    private List<String> products;
    private String orderNumber;

    public OrderCreatedEventV1(){}

    public List<String> getProducts() {
        return products;
    }

    public void setProducts(List<String> products) {
        this.products = products;
    }

    public String getOrderNumber() {
        return orderNumber;
    }

    public void setOrderNumber(String orderNumber) {
        this.orderNumber = orderNumber;
    }
}
